src/main/java/calculator/domain/InputValidator.java

```diff
@@ -0,0 +1,43 @@
+package calculator.domain;
+
+public class InputValidator {
+
+    public static void validatePositiveNumber(int number) {
+        if (number <= 0) {
+            throw new IllegalArgumentException("Number must be positive");
+        }
+    }
+
+
+    public static void validateOnlyDigits(String input) {
+        try {
+            Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(
+                    "Input must be a string consisting of a delimiter and a positive number.");
+        }
+    }
+
+
+    public static void validateExtractedNumbers(String[] numbers) {
+        for (String number : numbers) {
+            validateOnlyDigits(number);
+            validatePositiveNumber(Integer.parseInt(number));
+        }
+    }
+
+    public static void validateCustomDelimiter(String customDelimiter) {
+        if (isOnlyDigits(customDelimiter)) {
+            throw new IllegalArgumentException("Custom delimiter must contain a string");
+        }
+    }
+
+    public static boolean isOnlyDigits(String input) {
+        try {
+            Integer.parseInt(input);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
```

Comment:
```suggestion
    public static void validateOnlyDigits(String input) {
        try {
            Integer.parseInt(input);
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException("Custom delimiter must contain a string");
        }
    }
```

이렇게 작성하는 건 어떨까요? 메서드 분리까지는 필요 없을 것 같아서요! 😄

Reply:
확실히 보기 편하네요! 어디까지 나눠야 하는 지에 대한 기준을 생각해봐야겠습니다.
그리고 깃허브에 이런기능도 있군요. 깃허브도 좀 더 공부해야겠습니다 ㅎㅎ..

Reply:
PR의 Files Changed에 들어가서 코드 블럭 잡고 
<img width="38" alt="image" src="https://github.com/user-attachments/assets/f5238023-3a39-4cec-8bf0-a732315b228e">
이 버튼 누르시면 돼요! ㅎ ㅎ